### ISSUE TITLE
[Bug]: controlnet模型导出报错Floating point exception (core dumped)

### ISSUE THREAD

**Issue by hnsywangxin** [PaddlePaddle/PaddleNLP#5577]
### 软件环境

```Markdown
paddle-bfloat            0.1.7
paddle2onnx              1.0.6
paddlefsl                1.1.0
paddlenlp                2.5.2.post0
paddlepaddle-gpu         2.4.2.post117
```

### 重复问题

- [X] I have searched the existing issues

### 错误描述

```Markdown
W0408 07:28:44.481940 35020 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 7.5, Driver API Version: 12.0, Runtime API Version: 11.7
W0408 07:28:44.483088 35020 gpu_resources.cc:91] device: 0, cuDNN Version: 8.7.
Fetching 18 files: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 18/18 [00:00<00:00, 10842.66it/s]
[2023-04-08 07:28:53,635] [    INFO] - loading configuration file /root/.cache/paddlenlp/ppdiffusers/runwayml/stable-diffusion-v1-5/text_encoder/config.json
[2023-04-08 07:28:53,638] [    INFO] - Model config CLIPTextConfig {
  "_name_or_path": "openai/clip-vit-large-patch14",
  "architectures": [
    "CLIPTextModel"
  ],
  "attention_dropout": 0.0,
  "bos_token_id": 0,
  "dropout": 0.0,
  "eos_token_id": 2,
  "hidden_act": "quick_gelu",
  "hidden_size": 768,
  "initializer_factor": 1.0,
  "initializer_range": 0.02,
  "intermediate_size": 3072,
  "layer_norm_eps": 1e-05,
  "max_position_embeddings": 77,
  "model_type": "clip_text_model",
  "num_attention_heads": 12,
  "num_hidden_layers": 12,
  "pad_token_id": 1,
  "paddlenlp_version": null,
  "projection_dim": 512,
  "return_dict": true,
  "torch_dtype": "float32",
  "transformers_version": "4.21.0.dev0",
  "vocab_size": 49408
}

cross_attention_dim: 768
 unet_in_channels: 4
 vae_encoder_in_channels: 3
 vae_decoder_latent_channels: 4
Save text_encoder model in /opt/model/paddle_onnx_trt_control_sd15_canny/text_encoder/inference successfully.
Save unet model in /opt/model/paddle_onnx_trt_control_sd15_canny/unet/inference successfully.
Save vae_encoder model in /opt/model/paddle_onnx_trt_control_sd15_canny/vae_encoder/inference successfully.
Save vae_decoder model in /opt/model/paddle_onnx_trt_control_sd15_canny/vae_decoder/inference successfully.
WARNING: Logging before InitGoogleLogging() is written to STDERR
W0408 07:30:18.869315 35020 analysis_config.cc:972] It is detected that mkldnn and memory_optimize_pass are enabled at the same time, but they are not supported yet. Currently, memory_optimize_pass is explicitly disabled
Floating point exception (core dumped)
```

### 稳定复现步骤 & 代码

在controlnet/export_model.py中的代码
`fastdeploy_pipeline = FastDeployStableDiffusionControlNetPipeline(
        vae_encoder=FastDeployRuntimeModel.from_pretrained(output_path/"vae_encoder"),
        vae_decoder=FastDeployRuntimeModel.from_pretrained(output_path/"vae_decoder"),
        text_encoder=FastDeployRuntimeModel.from_pretrained(output_path/"text_encoder"),
        unet=FastDeployRuntimeModel.from_pretrained(output_path/"unet"),
        tokenizer=pipeline.tokenizer,
        scheduler=pipeline.scheduler,
        safety_checker=None,
        feature_extractor=None,
        requires_safety_checker=False,
    )`通不过，报错Floating point exception (core dumped)

**Comment by JunnYu** [comment]
为了能顺利导出模型，可能当前需要你安装develop版本paddlepaddle和develop的fastdeploy
```sh
python -m pip install paddlepaddle-gpu==0.0.0.post117 -f https://www.paddlepaddle.org.cn/whl/linux/gpu/develop.html
pip install fastdeploy-gpu-python==0.0.0 -f https://www.paddlepaddle.org.cn/whl/fastdeploy_nightly_build.html
```

**Comment by hnsywangxin** [comment]
> develop

新版本的都试过了，也会报错

**Comment by JunnYu** [comment]
你好，我这里今天（4月10日）试了一下develop的paddle和fastdeploy能够正确导出及使用。
![image](https://user-images.githubusercontent.com/50394665/230814391-daf9153b-054c-4934-915e-6a4c77d913c1.png)

**Comment by hnsywangxin** [comment]
@JunnYu 非常感谢，我全部将依赖库重新安装了一次，tensorrt终于可以正常跑了，但是我用的backend是tensorrt的时候，显存占用比正常的要多跟多（diffusers官方代码[StableDiffusionControlNetPipeline](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py#L48:~:text=EXAMPLE_DOC_STRING%20%3D%20%22%22%22-,Examples,-%3A)显存约4G，paddle tensorrt约7G）,不过速度上由2.6s降低到了1.5s,硬件设备为rtx 8000
另外，我不用tensorrt时仍然会报段错误
<img width="951" alt="image" src="https://user-images.githubusercontent.com/18531174/230825373-aa89507c-0166-401a-b4de-5fe5b0db69bc.png">

**Comment by JunnYu** [comment]
对，当前需要使用`backend="paddle_tensorrt"`，如狗使用`backend="paddle"`的话，目前会存在问题。

**Comment by JunnYu** [comment]
还有想问一下，你说的 （diffusers官方代码[StableDiffusionControlNetPipeline](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py#L48:~:text=EXAMPLE_DOC_STRING%20%3D%20%22%22%22-,Examples,-%3A)显存约4G，paddle tensorrt约7G） 这个能够提供一下测试代码吗？然后我们这里看一下

**Comment by hnsywangxin** [comment]
> 还有想问一下，你说的 （diffusers官方代码[StableDiffusionControlNetPipeline](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py#L48:~:text=EXAMPLE_DOC_STRING%20%3D%20%22%22%22-,Examples,-%3A)显存约4G，paddle tensorrt约7G） 这个能够提供一下测试代码吗？然后我们这里看一下

非常感谢，代码大致如下
```import cv2
from PIL import Image
import numpy as np
import torch
from diffusers import StableDiffusionPipeline, StableDiffusionControlNetPipeline, ControlNetModel
from diffusers import UniPCMultistepScheduler,DPMSolverMultistepScheduler
from controlnet_aux import HEDdetector
from diffusers.utils import load_image

seed = 1000
generator = torch.Generator('cuda:2').manual_seed(seed)
controlnet = ControlNetModel.from_pretrained('lllyasviel/sd-controlnet-hed', torch_dtype=torch.float16).to('cuda:2'),
pipe_local = StableDiffusionControlNetPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", controlnet=controlnet, torch_dtype=torch.float16).to('cuda:2')
pipe_local.scheduler = UniPCMultistepScheduler.from_config(pipe_local.scheduler.config)

hed = HEDdetector.from_pretrained('lllyasviel/ControlNet')
image = load_image('you image')
canny_image = np.array(image)
low_threshold = 100
high_threshold = 200
canny_image = cv2.Canny(canny_image, low_threshold, high_threshold)
zero_start = canny_image.shape[1] // 4
zero_end = zero_start + canny_image.shape[1] // 2
canny_image[:, zero_start:zero_end] = 0
canny_image = canny_image[:, :, None]
canny_image = np.concatenate([canny_image, canny_image, canny_image], axis=2)
canny_image = Image.fromarray(canny_image)
hed_image = hed(image)
images = [canny_image, hed_image]
prompt = 'interior design,C4D, rendering, Vray, 3D MAX'
negative_prompt = 'soft, blurry, poor quality, text, type, logo ,watermark, oversharp, painting, brush strokes, illustration, saturated, high contrast'

with torch.inference_mode():
    images = pipe_local(prompt, image=[image], guidance_scale=7, num_inference_steps=20, num_images_per_prompt=1,
                        height=512, width=512, generator=generator).images[0]```

**Comment by ViewWholeWorld** [comment]
> 对，当前需要使用`backend="paddle_tensorrt"`，如狗使用`backend="paddle"`的话，目前会存在问题。

您好，我在用 fastdeploy 部署 controlnet 的时候，用了 paddle_tensorrt的 backend，但是内存占用特别大，在加载用了 13G 的时候，OOM了，请问下这个需要多少内存多少显存才能完成加载？

**Comment by hnsywangxin** [comment]
@ViewWholeWorld 我这个没有那么夸张，大概6-7G吧

**Comment by zhangvia** [comment]
@JunnYu 


> @JunnYu 非常感谢，我全部将依赖库重新安装了一次，tensorrt终于可以正常跑了，但是我用的backend是tensorrt的时候，显存占用比正常的要多跟多（diffusers官方代码[StableDiffusionControlNetPipeline](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py#L48:~:text=EXAMPLE_DOC_STRING%20%3D%20%22%22%22-,Examples,-%3A)显存约4G，paddle tensorrt约7G）,不过速度上由2.6s降低到了1.5s,硬件设备为rtx 8000 另外，我不用tensorrt时仍然会报段错误 <img alt="image" width="951" src="https://user-images.githubusercontent.com/18531174/230825373-aa89507c-0166-401a-b4de-5fe5b0db69bc.png">
你是导出controlnet后用controlnet_infer推理的吗，我运行推理时会卡在loading shape这里
![image](https://user-images.githubusercontent.com/38352569/232014976-b39f883b-f9c6-4075-9812-38f483016258.png)
有人知道为啥吗。。。

**Comment by hnsywangxin** [comment]
已经实现用tensorrt加速controlnet，效率不错，可参考repo：https://github.com/hnsywangxin/controlnet_stable_tensorrt
